File: osmosm7/shortest_path/11265_끝나지않는파티.py
import sys
input = sys.stdin.readline
n,m = map(int,input().split())
mat = [list(map(int,input().split())) for _ in range(n)]
for k in range(n):
    for i in range(n):
        for j in range(n):
            mat[i][j] = min(mat[i][j],mat[i][k] + mat[k][j])
for _ in range(m):
    s,e,t = map(int,input().split())
    if mat[s-1][e-1] <= t:
        print('Enjoy other party')
    else:
        print('Stay here')

File: .archive/dkswndms4782/shortest_path/11265_끝나지않는파티.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int INF = 10000000;

int main(){
    int N,M,A,B,C; cin >> N >> M;
    int time[501][501];
    for(int i = 0;i<N;i++) for(int j = 0;j<N;j++) cin >> time[i][j];
    // k : 거쳐가는 정점, i : 출발 정점, j : 도착 정점
    for(int k=0;k<N;k++) for(int i=0;i<N;i++) for(int j=0;j<N;j++) time[i][j] = min(time[i][j], time[i][k] + time[k][j]);
    while(M--){
        cin >> A >> B >> C;
        if(time[A-1][B-1] <= C) cout << "Enjoy other party\n";
        else cout << "Stay here\n";
    }
}